plugin.rb

```diff
@@ -209,8 +209,8 @@ class ::ListController
     ::TopicAssigner.auto_assign(post, force: true)
   end
 
-  on(:topic_closed) do |topic|
-    if SiteSetting.unassign_on_close
+  on(:topic_status_updated) do |topic, status|
+    if SiteSetting.unassign_on_close && (status == 'closed' || status == 'autoclosed')
```

Comment:
Can also be written as ;)

```suggestion
    if SiteSetting.unassign_on_close && status["closed"]
```